### PR TITLE
[Code Coverage] Improve coverage of stream_writable.js

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
Removed the if check, since it would never be called, as doWrite only happens when state.writing is false.
IoT.js-DCO-1.0-Signed-off-by: Daniel Balla dballa@inf.u-szeged.hu

### CODE REVIEW

File: test/run_pass/test_stream_writable.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+var assert = require('assert');
+
+var Writable = require('stream').Writable;
+var writable = new Writable({highWaterMark: 0});
+
+var writable2 = Writable({highWaterMark: 0});
+
+assert.throws(function() { writable._write(0, 0, 0); }, Error);
+
+process.platform = 'nuttx';
```
Comment (hs0225) [inline test/run_pass/test_stream_writable.js]: Why did you change `process.platform`?

**Comment by yichoi** [comment]
@DanielBallaSZTE please rebase it to the master

File: test/run_pass/test_stream_writable.js (hunk 1 above)
Comment (DanielBallaSZTE) [inline test/run_pass/test_stream_writable.js]: Wanted to test if the EOF event runs without errors, making the process believe it's nuttx, so the code runs.

Comment (DanielBallaSZTE) [inline test/run_pass/test_stream_writable.js]: Nevertheless, I don't think it's needed, so I'm removing it, just left it there.

**Comment by DanielBallaSZTE** [comment]
@yichoi Rebased, however the master build fails.

**Comment by daeyeon** [comment]
We should avoid adding tests for private/protected members since it make tests verbose and make sources much harder to maintain.

This test doesn't give exact information about what are tested. A common pattern for unit test in my mind is AAA (Arrange, Act, Assert). In this code, I can find sections to arrange values and act testing. However, the section for assert, which verifies that the action behaves as expected, are not enough. This is a common problem I can find these days from commits related to code coverage.

**Comment by yichoi** [comment]
Closing

**Comment by DanielBallaSZTE** [comment]
@daeyeon These tests are to cover the uncovered branches, which point to nothing, like in line 261 the else case. There's nothing to assert.

**Comment by daeyeon** [comment]
Despite your explanation, the test still looks like a cheat to just pass the uncovered branches, and they are unclear. If it's impossible to cover them via normal operations, that may mean we need to consider refactoring it.
